File: Source/Utility/FilterList.cs
using WCSharp.Api;
using static WCSharp.Api.Common;

public static class FilterList
{
    public static filterfunc KittyFilter { get; private set; } = Utility.CreateFilterFunc(() => GetFilterUnit().UnitType == Constants.UNIT_KITTY);
    public static filterfunc DogFilter { get; private set; } = Utility.CreateFilterFunc(() => GetFilterUnit().UnitType == Constants.UNIT_CUSTOM_DOG);
    public static boolexpr IssuedOrderAtkOrder = Condition(() => GetIssuedOrderId() == WolfPoint.AttackOrderID);
    public static boolexpr IssuedOrderStopOrder = Condition(() => GetIssuedOrderId() == WolfPoint.StopOrderID);
    public static boolexpr IssuedOrderMoveOrder = Condition(() => GetIssuedOrderId() == WolfPoint.MoveOrderID);
    public static boolexpr IssuedOrderHoldPosiiton = Condition(() => GetIssuedOrderId() == WolfPoint.HoldPositionOrderID);
    public static boolexpr UnitTypeWolf = Condition(() => GetTriggerUnit().UnitType == Wolf.WOLF_MODEL);
}

File: src/Utility/FilterList.ts


class FilterList
{
    public static KittyFilter: filterfunc = Utility.CreateFilterFunc(()  { return GetFilterUnit().UnitType == Constants.UNIT_KITTY); }
    public static KittyFilterOrShadow: filterfunc = Utility.CreateFilterFunc(()  { return GetFilterUnit().UnitType == Constants.UNIT_KITTY || GetFilterUnit().UnitType == Constants.UNIT_SHADOWKITTY_RELIC_SUMMON); }
    public static DogFilter: filterfunc = Utility.CreateFilterFunc(()  { return GetFilterUnit().UnitType == Constants.UNIT_CUSTOM_DOG); }
    public static IssuedOrderAtkOrder: boolexpr = Condition(()  { return GetIssuedOrderId() == WolfPoint.AttackOrderID); }
    public static IssuedOrderStopOrder: boolexpr = Condition(()  { return GetIssuedOrderId() == WolfPoint.StopOrderID); }
    public static IssuedOrderMoveOrder: boolexpr = Condition(()  { return GetIssuedOrderId() == WolfPoint.MoveOrderID); }
    public static IssuedOrderHoldPosiiton: boolexpr = Condition(()  { return GetIssuedOrderId() == WolfPoint.HoldPositionOrderID); }
    public static UnitTypeWolf: boolexpr = Condition(()  { return GetTriggerUnit().UnitType == Wolf.WOLF_MODEL); }
}
